service/src/java/org/apache/hive/service/cli/operation/SQLOperation.java

```diff
@@ -451,7 +451,8 @@ public void close() throws HiveSQLException {
   public TableSchema getResultSetSchema() throws HiveSQLException {
     // Since compilation is always a blocking RPC call, and schema is ready after compilation,
     // we can return when are in the RUNNING state.
-    assertState(Arrays.asList(OperationState.RUNNING, OperationState.FINISHED));
+    assertState(Arrays.asList(OperationState.RUNNING, OperationState.FINISHED,
```

Comment:
I might be coming in from the other direction - but can't we remove `RUNNING` as well?
as a matter of fact I think this method should only be called in "final" state - and the only final state in which this method makes sense is when the state is `FINISHED`

from where this is being called when it causes the issue?